mcuboot: Add MCUBOOT hw rollback protection (#9196)
nrfconnect/sdk-nrf: subsys/bootloader/image/CMakeLists.txt

  assert(CONFIG_FW_INFO "CONFIG_FW_INFO must be set")
endif()

if(NOT CONFIG_BOOTLOADER_MCUBOOT)
# When MCUBOOT is enabled we need to execute the provision_hex.cmake file
# after the MCUBOOT child image is built since depending on it's
# configuration we might need to provision data for MCUBOOT as well
# (montonic counters).

> include(../cmake/provision_hex.cmake)

Reviewer: indentation

> include(../cmake/provision_hex.cmake)

Reviewer: NB: I'm pretty sure provision_hex.cmake will always be executed before the MCUBoot child image is built. Perhaps it was meant to say "provision_hex.cmake must be executed after MCUBoot has been configured."

It's not clear to me what the actual dependency is, but the explanation looks wrong to me.

Can we move this to somewhere after "the child image is built" ?

And do something like:

if(CONFIG_PROVISION_HEX)
include(path/to//provision_hex.cmake)
endif()

Then it would only be executed from one place and it would be more clear the condition for when provision_hex.cmake is needed.

> # This creates the provision.hex file when invoked. When MCUBOOT
> # is enabled we need to invoke this file after the MCUBOOT child
> # image is built since depending on it's configuration we might
> # need to provisiong data for MCUBOOT as well (security counters).
> include(${CMAKE_CURRENT_LIST_DIR}/../cmake/provision_hex.cmake)

Reviewer: ```suggestion
  # Let MCUBoot execute provision_hex.cmake when MCUBoot is enabled
  # since it may need to provision security counters.
  include(../cmake/provision_hex.cmake)
```

Author: I will change the path and I will refactor the text a bit. I want the information about the order that this file is included in the description because I value it. So I will not remove that.
